packages/@aws-cdk/aws-apigatewayv2/lib/websocket/stage.ts

```diff
@@ -87,14 +112,103 @@ export class WebSocketStage extends StageBase implements IWebSocketStage {
     this.api = props.webSocketApi;
     this.stageName = this.physicalName;
 
+    /**
+     * In CfnStage the rate limits, the data trace logging control and the detailed
+     * metrics are all lumped together in the default route settings, but they're
+     * conceptually orthogonal.  I'm following the pattern that was started by
+     * breaking out the rate limits to break out the conceptually distinct
+     * throttle settings, data trace logging settings and access log settings.
+     */
+    const rateLimits = !props.throttle ? undefined : {
+      throttlingBurstLimit: !props.throttle.burstLimit ? undefined : props.throttle.burstLimit,
+      throttlingRateLimit: !props.throttle.rateLimit ? undefined : props.throttle.rateLimit,
+    };
+
+    /**
+     * The data trace log settings for CfnStage use two parameters, dataTraceEnabled to
+     * turn on the logging and a logging level.  However, specifing a logging level of
+     * OFF is the same as disabling data tracing and setting it to INFO or ERROR implies
+     * that data tracing is on, so the value of the logging flag is being deduced from
+     * the value of the log level.
+     */
+    const dataTraceLoggingSettings =
+      (!props.dataTraceLoggingLevel || props.dataTraceLoggingLevel === 'OFF') ?
+        undefined :
+        {
+          dataTraceEnabled: true,
+          loggingLevel: props.dataTraceLoggingLevel,
+        };
+
+    // If the user has specfied a retention time and/or removal policy for the log group use it,
+    // otherwise make it a 30 day retention and a removal policy of DESTROY.
+    if (props.dataTraceLoggingLevel && props.dataTraceLoggingLevel !== 'OFF') {
+      const retentionInDays = !props.dataTraceLogRetention ? 30 : Math.max(props.dataTraceLogRetention.toDays(), 1);
+
+      // There is no parameter for CfnStage that will accept a non-default log group, however
+      // if the log group is created before API Gateway tries to create a log entry for a data trace
+      // log it'll use the log group that this creates.
+      const dataLoggingLogGroup = new CfnLogGroup(this, 'data-trace-logging-log-group', {
```

Comment:
This won't work for existing API gateways.

Reply:
I see your point.  I don't see a way that this can be made to work in general, so I took out the retention time and retention policy parameters.